client/lib/site/index.js

```diff
@@ -100,6 +100,16 @@ Site.prototype.replace = function( attributes ) {
 
 	let changed = false;
 
+	// Set attrs which will get computed and will mean the same before doing equality checks
+	// so the 'change' event is not triggered. See Site.prototype.updateComputedAttributes.
+	if ( attributes.hasOwnProperty( 'options' ) ) {
+		let options = attributes.options;
+
+		if ( ! options.hasOwnProperty( 'default_post_format' ) || options.default_post_format === '0' ) {
+			options.default_post_format = 'standard';
+		}
+	}
+
```

Comment:
That makes sense, thanks. However, when I tried it (back then), it seems like the `computedAttrsCustomizer` is not even run if the `options` don't include the `default_post_format` prop. Maybe lodash assumes that if the property is not on one side of the compared objects, it is not equal. 

While I can look into lodash's source code to verify this, I like my current solution more (should be faster as we don't use our own `customizer` which has to be called on each prop). Or do you think this `customizer` approach (if worked) would be more preferable?